src/utils/hooks/useGetReviewStructureForSection/helpers.tsx

```diff
@@ -46,33 +67,34 @@ const generateReviewStructure: GenerateReviewStructure = ({
   // mutate fullApplicationStructure
   let newStructure: FullStructure = cloneDeep(fullApplicationStructure)
 
+  const { reviewQuestionAssignments, level, review } = reviewAssignment
+
   // This is usefull for linking assignments to elements
   newStructure = addElementsById(newStructure)
   newStructure = addSortedSectionsAndPages(newStructure)
 
-  newStructure = addIsAssigned(newStructure, reviewAssignment.reviewQuestionAssignments)
+  newStructure = addIsAssigned(newStructure, reviewQuestionAssignments)
 
   // here we add responses from other review (not from this review assignmnet)
 
   setIsNewApplicationResponse(newStructure)
 
-  if ((data.reviewAssignment?.reviews?.nodes?.length || 0) > 1)
-    console.error(
-      'More then one review associated with reviewAssignment with id',
-      reviewAssignment.id
-    )
-  // There will always just be one review assignment linked to a review. (since review is related to reviewAssignment, many to one relation is created)
-  const reviewResponses = data?.reviewAssignment?.reviews?.nodes[0]?.reviewResponses.nodes
-  if (reviewResponses) {
-    newStructure = addThisReviewResponses({
-      structure: newStructure,
-      sortedReviewResponses: reviewResponses as ReviewResponse[], // Sorted in useGetReviewResponsesQuery
-    })
-  }
+  // thisReviewLatestResponse and thisReviewPreviousResponse
+  // latestPreviousLevelReviewResponse and previousPreviousReviewLevelResponse
+  // latestOriginalReviewResponse and previousOriginalReviewResponse
+  newStructure = addAllReviewResponses(newStructure, data)
+
   // review info comes from reviewAssignment that's passed to this hook
-  newStructure.thisReview = reviewAssignment.review
+  newStructure.thisReview = review
+
+  newStructure = addIsPendingReview(newStructure, level)
+  newStructure = addIsActiveReviewResponse(newStructure)
 
-  generateReviewProgress(newStructure)
+  if (level === 1) {
+    generateReviewProgress(newStructure)
+  } else {
+    generateReviewConsolidationProgress(newStructure)
```

Comment:
Simply `generateConsolidationProgress` would be better. And as discussed in person, perhaps create the progress in a separated field - since it has all different fields. On `FullStructure` can be also `consolidationProgress`.

Reply:
agree!
53e1d3efc5d0567881630a2606d04bd7e3122bf1